components/datetime/src/skeleton.rs

```diff
@@ -195,6 +200,62 @@ impl TryFrom<&str> for Skeleton {
     }
 }
 
+/// Convert a Pattern into a Skeleton. This will remove all of the string literals, and sort
+/// the fields into the canonical sort order. Not all fields are supported by Skeletons, so map
+/// fields into skeleton-appropriate ones. For instance, in the "ja" locale the pattern "aK:mm"
+/// gets transformed into the skeleton "hmm".
+impl From<&Pattern> for Skeleton {
```

Comment:
Question: Hmm. Can you help me find where this impl is being used? I'm a bit concerned about exposing an API that may or may not be best practice.

Reply:
This is being used in the transform with: `apply_coarse_hour_cycle`.

`let skeleton = skeleton::Skeleton::from(&pattern);`

Reply:
This is now hidden behind `provider_transform_utils`.